src/mlpack/methods/ann/init_rules/network_init.hpp

```diff
@@ -52,7 +52,8 @@ class NetworkInitialization
    * @param network Network that should be initialized.
    * @param parameter The network parameter.
    */
-  void Initialize(const std::vector<LayerTypes>& network, arma::mat& parameter)
+  void Initialize(const std::vector<LayerTypes>& network, arma::mat& parameter,
+      size_t parameterOffset = 0)
   {
     // Determine the number of parameter/weights of the given network.
     size_t weights = 0;
```

Comment:
We should do the following only if ``parameter.is_empty()`` is true, that way we use the specified ``parameter`` matrix and don't allocated a new one every time we call ``Initialize``.